src/pages/components/loadingspinner.tsx

```diff
@@ -0,0 +1,320 @@
+          text="Loading"
+        />
+      </ComponentExample>
+    </ComponentDescription>
+
+    <NoteBox title={content['note.title']} items={content['note.items']} />
+
+    {content.sections.map((section, index) => (
+      <Section
+        key={index}
+        mainTitle={section.title}
+        paragraphs={section.paragraphs}
+        links={section.links}
+      />
+    ))}
+
+    <BasicExample
+      title={content['example.basic.heading']}
+      desc={content['example.basic.description']}
+      noCode={true}
+    />
+
+    <SuccessExample
+      title={content['example.success.heading']}
+      desc={content['example.success.description']}
+      noCode={true}
+    />
+
+    <FailingExample
+      title={content['example.error.heading']}
+      desc={content['example.error.description']}
+      noCode={true}
+    />
+  </Layout>
+);
+
+const BasicExample = ({
+  title,
+  desc,
+  noCode,
+  ...passProps
+}: Partial<LoadingSpinnerProps> & {
+  title: string;
+  desc: string;
+  noCode: boolean;
+}): JSX.Element => {
+  const [visible, setVisible] = useState(false);
+  const [loaded, setLoaded] = useState(0);
+  const [status, setStatus] = useState<LoadingSpinnerStatus>('loading');
+
+  const runLoader = (): void => {
+    let progress = 0;
+    const id = setInterval(frame, 300);
+    function frame(): void {
+      if (progress >= 100) {
+        clearInterval(id);
+        setVisible(false);
+        progress = 0;
+      } else {
+        progress = progress + 10;
+      }
+      setLoaded(progress);
+    }
+  };
+
+  return (
+    <ComponentDescription mainTitle={title} description={desc} noCode={noCode}>
+      <ComponentExample
+        style={{
+          marginBottom: defaultSuomifiTheme.spacing.s,
+          display: 'flex',
+          flexDirection: 'column',
+        }}
+      >
+        <Button
+          disabled={visible}
+          onClick={() => {
+            setStatus('loading');
+
+            if (!visible) {
+              runLoader();
+            }
+            setVisible(!visible);
+          }}
+        >
+          {content['example.startButton']}
+        </Button>
+
+        <div
+          aria-live="assertive"
+          aria-busy={status === 'loading'}
+          style={{
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'center',
+            width: '100%',
+            height: '100px',
+            marginTop: defaultSuomifiTheme.spacing.m,
+            border: `1px solid ${defaultSuomifiTheme.colors.depthLight1}`,
+          }}
+        >
+          {visible && (
+            <LoadingSpinner
+              status={status}
+              text={
+                status !== 'success'
+                  ? content['example.loading.active']
+                  : content['example.loading.success']
+              }
+            />
+          )}
+        </div>
+      </ComponentExample>
+    </ComponentDescription>
+  );
+};
+
+const SuccessExample = ({
```

Comment:
https://user-images.githubusercontent.com/17459942/173033635-4d10eb3a-f5f8-43d0-9aa7-3593d7e0ba7e.mov

I would probably just have the success and failing examples follow the same operational logic from the first example, like this.

Having the button say "Peruuta lataus" which then removes the component from view is a bit weird in my opinion